src/exec/spawn_command.c

```diff
@@ -94,12 +94,14 @@ int	spawn_command(t_ast *node, t_ctx *ctx, t_pipeline_conf *conf)
 	else if (pid == 0)
 	{
 		argv = convert_cmd_args_to_array(node->cmd_args);
-		if (!handle_io(conf, node->redirects, ctx))
+		if (!handle_io(conf, node->redirects, ctx, false))
 			exit(EXIT_FAILURE);
 		reset_signal_handlers();
 		execute_command_internal(argv, ctx);
 	}
-	if (conf->fd_out == STDOUT_FILENO)
-		wait_for_children(pid, ctx);
+	if (conf)
+		close_pipe_fd(conf);
```

Comment:
I think this `close_pipe_fd()` is sufficient here and does not need same func in `execute_pipeline.c`

Reply:
This line ensures that unnecessary pipe file descriptors are closed in the parent process before executing external commands.

Reply:
Yes, this line needs to be here.